engine/src/exec-interface-image.cpp

```diff
@@ -313,46 +318,75 @@ void MCImage::GetFormattedWidth(MCExecContext& ctxt, integer_t& r_width)
 
 void MCImage::GetText(MCExecContext& ctxt, MCDataRef& r_text)
 {
-	recompress();
-	if (m_rep == nil || m_rep->GetType() == kMCImageRepReferenced)
-	{
-		r_text = MCValueRetain(kMCEmptyData);
-		return;
-	}
-	else
+    
+    
+    if (m_rep == nullptr || m_rep->GetType() == kMCImageRepReferenced)
+    {
+        r_text = MCValueRetain(kMCEmptyData);
+        return;
+    }
+    
+    bool t_success = false;
+    
+    
+    MCImage* t_image_to_work_on = this;
+    
+    if (m_rep && m_rep->IsLocked())
+    {
+        t_image_to_work_on = new MCImage(*this);
+    }
+    else
+    {
+        t_image_to_work_on->recompress();
+    }
+    
+    void *t_data = nil;
+	uindex_t t_size = 0;
+	if (t_image_to_work_on->m_rep->GetType() == kMCImageRepResident)
+		static_cast<MCResidentImageRep*>(t_image_to_work_on->m_rep)->GetData(t_data, t_size);
+	else if (t_image_to_work_on->m_rep->GetType() == kMCImageRepVector)
+		static_cast<MCVectorImageRep*>(t_image_to_work_on->m_rep)->GetData(t_data, t_size);
+	else if (t_image_to_work_on->m_rep->GetType() == kMCImageRepCompressed)
 	{
-		void *t_data = nil;
-		uindex_t t_size = 0;
-		if (m_rep->GetType() == kMCImageRepResident)
-			static_cast<MCResidentImageRep*>(m_rep)->GetData(t_data, t_size);
-		else if (m_rep->GetType() == kMCImageRepVector)
-			static_cast<MCVectorImageRep*>(m_rep)->GetData(t_data, t_size);
-		else if (m_rep->GetType() == kMCImageRepCompressed)
+		MCImageCompressedBitmap *t_compressed = nil;
+		t_compressed = static_cast<MCCompressedImageRep*>(t_image_to_work_on->m_rep)->GetCompressed();
+		if (t_compressed->data != nil)
 		{
-			MCImageCompressedBitmap *t_compressed = nil;
-			t_compressed = static_cast<MCCompressedImageRep*>(m_rep)->GetCompressed();
-			if (t_compressed->data != nil)
-			{
-				t_data = t_compressed->data;
-				t_size = t_compressed->size;
-			}
-			else
-			{
-				t_data = t_compressed->planes[0];
-				t_size = t_compressed->plane_sizes[0];
-			}
+			t_data = t_compressed->data;
+			t_size = t_compressed->size;
+		}
+		else
+		{
+			t_data = t_compressed->planes[0];
+			t_size = t_compressed->plane_sizes[0];
 		}
-
-		if (MCDataCreateWithBytes((const byte_t *)t_data, t_size, r_text))
-			return;
 	}
 	
-	ctxt . Throw();
+    if (MCDataCreateWithBytes((const byte_t *)t_data, t_size, r_text))
+    {
+        t_success = true;
+    }
+
+    if (t_image_to_work_on != this)
+    {
+        delete t_image_to_work_on;
+    }
+    if (!t_success)
+    {
+        ctxt . Throw();
+        
+    }
 }
 
+
 void MCImage::SetText(MCExecContext& ctxt, MCDataRef p_text)
 {
-	bool t_success = true;
+    if (m_rep && m_rep->IsLocked())
+    {
+        ctxt . LegacyThrow(EE_IMAGE_MUTABLELOCK);
+        return;
+    }
+    bool t_success = true;
```

Comment:
I suggest adding a new line here, to separate out the early termination clauses from the body of the function.